### DISCUSSION TITLE
Set environment variables linked with branches from outside of workflow script

### DISCUSSION THREAD

**Discussion by taichi-ishitani** [community/community#26203] [Actions]
<p>I’d like to set environment variables linked with branches from outside of workflow script.</p>
<p>How can I do that?</p>
<p>Example:</p>
<p>FOO repositry depends on other respositries (BAR, BAZ) and I’d like to control which branch of BAR/BAZ will be checked out by using environment variables:</p>
<pre><code>$ cd bar; git checkout $BAR_BRANCH
$ cd baz; git chekcout $BAZ_BRANCH
</code></pre>
<p>When CI is invoked on the master branch I’d like to set both of $BAR_BRANCH and $BAZ_BRANCH to ‘master’.</p>
<p>When CI is invoked on a feature branch I’d like to set $BAR_BRANCH to ‘master’ and set $BAZ_BRANCH to ‘baz_new_feature’ branch.</p>
<p>I’d not like to put the above settings on the workflow script.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/taichi-ishitani">@taichi-ishitani</a> ,</p>
<p>Where do you want to set the environment variables linked with branches?</p>
<p>Currently, except the <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables" rel="nofollow noopener"><strong>default environment variables</strong></a> for each GitHub Actions workflow run are automatically set by GitHub, if you want to set custom environment variables, you need to specify the variables in the workflow file. So we have no any way to set custom environment variables outside of workflow script.</p>
<p>More details, you can reference the docs about <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables" rel="nofollow noopener"><strong>Using environment variables</strong></a>.</p>
<p>Of course if you project really need this feature, I recommend you directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by taichi-ishitani** [comment]
<p><a class="mention" href="/brightran">@brightran</a> ,</p>
<p>Thank you for your replay and I understood.</p>
<p>I will open a feature request ticket according to your advice.</p>
<p>Regards,</p>
<p>Taichi Ishitani</p>
